r/R/dplyr-funcs-datetime.R

```diff
@@ -168,3 +168,23 @@ register_bindings_datetime <- function() {
     build_expr("cast", x, options = list(to_type = date32()))
   })
 }
+
+binding_format_datetime <- function(x, format = "", tz = "", usetz = FALSE) {
+  if (usetz) {
+    format <- paste(format, "%Z")
+  }
+
+  if (call_binding("is.POSIXct", x)) {
+    # the casting part might not be required once
+    # https://issues.apache.org/jira/browse/ARROW-14442 is solved
+    # TODO revisit the steps below once the PR for that issue is merged
+    if (tz == "" && x$type()$timezone() != "") {
+      tz <- x$type()$timezone()
```

Comment:
I thought I had asked this before, but I can't find it in the comments, so I must be making it up!

Could you add a test down below for something like `arrow_table() %>% mutate(start_date = format(as.POSIXct("2022-01-01 01:01:00"))`

I have a suspicion that using `$type()$timezone()` will error with R input here.

Reply:
The evaluation for "regular" R objects goes via:
```r
    if (!inherits(x, "Expression")) {
      return(format(x, ...))
    }
```

so, that particular case doesn't get the chance to stumble on `$type()$timezone()`

Reply:
(that was back in the days when my logic was split between _Expression_ and _regular_ R object)

Reply:
I have added a unit test.